docs/sources/mimir/release-notes/v2.16.md

```diff
@@ -0,0 +1,83 @@
+In rulers, when rule concurrency is enabled for a rule group, its rules will now be reordered and run in batches based on their dependencies.
+This increases the number of rules that can potentially run concurrently.
+Note that the global and tenant-specific limits around the number of rule groups and rules per group still apply.
+
+Using `mimirtool` to analyze Grafana dashboards now supports bar chart, pie chart, state timeline, status history, histogram, candlestick, canvas, flame graph, geomap, node graph, trend, and XY chart panels.
+
+## Important changes
+
+In Grafana Mimir 2.16, the following behavior has changed:
+
+Starting this release Grafana Mimir only provides container images based on [distroless](https://github.com/GoogleContainerTools/distroless) image. The Alpine Linux based container images were deprecated in the 2.12 release and are no longer built.
+
+How experimental PromQL functions are enabled has changed.
+
+- The experimental CLI flags `-querier.promql-experimental-functions-enabled` and `-query-frontend.block-promql-experimental-functions` and respective YAML configuration have been removed from query-frontends and queriers.
+- Experimental PromQL functions are disabled by default but can be enabled using only the per-tenant setting `enabled_promql_experimental_functions`.
+
+Support for native histograms and out-of-order native histograms is enabled by default in ingesters.
+
+Distributors discard float and histogram samples with duplicated timestamps from each timeseries in a request before the request is forwarded to ingesters.
+Discarded samples are tracked by `cortex_discarded_samples_total` metrics with the reason `sample_duplicate_timestamp`.
+
+## Experimental features
+
+Grafana Mimir 2.16 includes some features that are experimental and disabled by default.
+Use these features with caution and report any issues that you encounter:
+
+Distributors now include experimental support for the Influx [line protocol](https://docs.influxdata.com/influxdb/cloud/reference/syntax/line-protocol/).
+
+Query-frontends now include experimental support to "spin off" subqueries as actual range queries, so that they benefit from query acceleration techniques such as sharding, splitting, and caching.
+
+## Bug fixes
+
+- Distributor: Use a boolean to track changes while merging the ReplicaDesc components, rather than comparing the objects directly. #10185
+- Querier: fix timeout responding to query-frontend when response size is very close to `-querier.frontend-client.grpc-max-send-msg-size`. #10154
+- Query-frontend and querier: show warning/info annotations in some cases where they were missing (if a lazy querier was used). #10277
+- Query-frontend: Fix an issue where transient errors are inadvertently cached. #10537 #10631
+- Ruler: fix indeterminate rules being always run concurrently (instead of never) when `-ruler.max-independent-rule-evaluation-concurrency` is set. https://github.com/prometheus/prometheus/pull/15560 #10258
```

Comment:
```suggestion
- Ruler: Fix indeterminate rules always, instead of never, running concurrently when `-ruler.max-independent-rule-evaluation-concurrency` is set. https://github.com/prometheus/prometheus/pull/15560 #10258
```